### PR TITLE
[kube-state-metrics] Refactor

### PR DESCRIPTION
**Pull request by zanac1986**
Signed-off-by: zanac1986 <zanhsieh@protonmail.com>

#### Which issue this PR fixes

Refactor following:

- Fix indention
- Reformat `templates/_helper.tpl` to close to the code similar to the one thru `helm create kube-state-metrics`
- Move pdb apiVersion to `templates/_helper.tpl`

No change to values.yaml, which means backward compatible.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-state-metrics/README.md [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
-## Get Repo Info
+## Get Repository Info
 
 ```console
 helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
 helm repo update
 ```
 
-_See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation._
+_See [helm repository](https://helm.sh/docs/helm/helm_repo/) for command documentation._
```
Comment (zanhsieh) [inline charts/kube-state-metrics/README.md:12]:
```suggestion
_See [`helm repo`](https://helm.sh/docs/helm/helm_repo/) for command documentation._
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Comment by mrueg** [comment]
@zanac1986 @zanhsieh are you approving your own changes here or is the name just a coincidence?

**Comment by zanhsieh** [comment]
@mrueg 
I can't merge it. Only you can.

**Comment by monotek** [comment]
That does not answer the question, which i like to get answered too.

**Comment by mrueg** [comment]
> @mrueg @monotek I think just coincidence. I don't know that guy.

What a coincidence! Thanks for clarifying. I was wondering because your emails did only differ in provider name as well as both of your github account names were "MH" :D

**Review by dotdc — CHANGES_REQUESTED** [review]
+1 @monotek 

For the PR itself, it is a lot of changes to test for limited benefits, and the risk of breaking something is high in my opinion. 

@zanac1986, If you want these changes to be merged, cut them into smaller PR's and explain why we should merge for each of them.

Do we agree to close this PR? 
@mrueg @monotek

**Pull request closed without merging**